lib/src/impl_ffi/impl_ffi.utils.dart

```diff
@@ -211,35 +217,38 @@ class _Scope {
       }
     }
   }
+
+  @override
+  void free(ffi.Pointer pointer) {
+    // Does nothing, use `release` instead.
+    // Not throwing, so that this can actually be used as an Allocator.
+  }
 }
 
-/// Invoke [fn] with [ffi.Pointer<T>] of size count and release the pointer
-/// when [fn] returns.
+/// Invoke [fn] with [p], and release [p] when [fn] returns.
 R _withAllocation<T extends ffi.NativeType, R>(
-  int count,
+  ffi.Pointer<T> p,
   R Function(ffi.Pointer<T>) fn,
 ) {
```

Comment:
This suddenly seems a lot less convenient when you have to type the Type being allocated twice..

Can't we keep the old signature? That way it's not so magic that you have yo use `_sslAlloc`.

I think this is a convenience method from before I implemented `_Scope`. Half the places using it should probably be ported to just using `_Scope`. On the other hand I guess it is convenient have this.

Writing: `_withAllocation(1, (ffi.Pointer<CBS> cbs) { ...`  is certainly nice..

Reply:
Let me see if I can rewrite the uses to something like:

```dart
  // In a pool multiple resources can be allocated, which will all be freed
  // at the end of the scope.
  using((Pool pool) {
    final p = pool<Int64>(2);
    final p2 = pool<Int64>(2);
    p[0] = 1;
    p[1] = 2;
    MemMove(p2.cast<Void>(), p.cast<Void>(), 2 * sizeOf<Int64>());
    Expect.equals(1, p2[0]);
    Expect.equals(2, p2[1]);
  });
```

https://github.com/dart-lang/sdk/blob/master/samples/ffi/resource_management/pool_sample.dart#L48-L58

The scope is essentially the same as the pool.

Reply:
Yeah, we could probably have two static methods `_Scope.using` and `_Scope.usingAsync`..

But really, don't sweat it over cleanup..

Reply:
Tracking in https://github.com/google/webcrypto.dart/issues/19.